core/domain/learner_progress_services.py

```diff
@@ -614,6 +830,163 @@ def _get_filtered_completed_exp_summaries(
     return filtered_completed_exp_summaries, nonexistent_completed_exp_ids
 
 
+def get_all_completed_story_ids(user_id):
+    """Returns a list with the ids of all the stories completed by the
+    user.
+
+    Args:
+        user_id: str. The id of the learner.
+
+    Returns:
+        list(str). A list of the ids of the stories completed by the
+        learner.
+    """
+
+    completed_activities_model = (
+        user_models.CompletedActivitiesModel.get(
+            user_id, strict=False))
+
+    if completed_activities_model:
+        activities_completed = _get_completed_activities_from_model(
+            completed_activities_model)
+
+        return activities_completed.story_ids
+    else:
+        return []
+
+
+def _get_filtered_completed_story_summaries(
+        user_id, story_summaries, story_ids):
+    """Returns a list of summaries of the completed story ids, the ids
+    of stories that are no longer present and the summaries of the
+    stories being shifted to the incomplete section on account of new
+    addition of nodes.
+
+    Args:
+        user_id: str. The id of the learner.
+        story_summaries: list(StorySummary). The list of story
+            summary domain objects to be filtered.
+        story_ids: list(str). The ids of the story corresponding to
+            the story summary domain objects.
+
+    Returns:
+        tuple. A 3-tuple whose elements are as follows:
+        - list(StorySummary). A filtered list with the summary domain
+            objects of the completed story.
+        - list(str). The ids of the stories that are no longer present.
+        - list(StorySummary). The summaries corresponding to those
+            stories which have been moved to the in progress section on
+            account of new nodes being added to them.
+    """
+    nonexistent_completed_story_ids = []
+    completed_to_incomplete_stories = []
+    filtered_completed_story_summaries = []
+
+    for index, story_summary in enumerate(story_summaries):
+        if story_summary is None:
+            nonexistent_completed_story_ids.append(story_ids[index])
+        else:
+            story_id = story_summary.id
+            story = story_fetchers.get_story_by_id(story_id)
+            if len(story_fetchers.get_completed_node_ids(
+                    user_id, story_id)) != len(story_summary.node_titles):
+                remove_story_from_completed_list(user_id, story_id)
+                mark_story_as_incomplete(user_id, story_id)
+                completed_to_incomplete_stories.append(story_summary)
+            elif not story_services.is_story_published_and_present_in_topic(
+                    story):
```

Comment:
So, even if the story exists and it is unpublished, we won't show it to the learner, right?

Reply:
Yeah we won't show it to the learner.